### DISCUSSION TITLE
Azure AD B2C MFA with TOTP using any Authenticator app

### DISCUSSION THREAD

**Discussion by charlesdays** [azure-ad-b2c/samples#577] [Q&A]
Greetings,
I've successfully used this sample to enable acess with MFA onto an authenticated portal.
I've split the TOTP enrolment from the TOTP MFA request in 2 policies.
In this portal, after user signs in, to enable MFA, he needs to navigate to a link where he clicks "Enable MFA". This will call the MFA enrollment Policy.
Issue here is when I call the policy from an icognito session, or outisde the authenticated portal session, the policy enrolment asks for authentication , presents the QR Code to enable the account in Authenticator and finally the OTP code to be entered.
When this policy is called within an authenticated session, user is not prompted for authentication of course, it presents the QR Code to add the account in Authenticator, but misses the OPT Code with immediate JWT token issued.
Iv'e tried all the tricks to break the SSO session but it continues not to prompt the code and register de Oauth device.

Q:Is the MFA provider looking at the existing authenticated session and has so bypassing the MFA request has it thinks there is already an existing session?

**Comment by wajihqidwai** [comment]
Hi Charles,
I am also facing the similar issue. Can you please share if you got the solution?
